tests/scripts/all.sh

```diff
@@ -2196,6 +2196,23 @@ support_build_baremetal () {
     ! grep -q -F time.h /usr/include/x86_64-linux-gnu/sys/types.h
 }
 
+support_build_warnings () {
+    type "$GCC_LATEST" >/dev/null 2>/dev/null
+    type "$CLANG_LATEST" >/dev/null 2>/dev/null
+}
+component_build_warnings () {
```

Comment:
Why are you adding a new component instead of adding more flags to the existing `test_build_opt`?

Reply:
Ah, I missed that. I'll combine them.